drivers/sensor/bh1750/bh1750.c

```diff
@@ -0,0 +1,153 @@
+/* Rohm BH1750 16 bit ambient light sensor
+ *
+ * Copyright (c) 2022 Lucas Heitzmann Gabrielli
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT rohm_bh1750
+
+#include <logging/log.h>
+#include <drivers/i2c.h>
+#include <drivers/sensor.h>
+
+#include "bh1750.h"
+
+LOG_MODULE_REGISTER(BH1750, CONFIG_SENSOR_LOG_LEVEL);
+
+static int bh1750_read(const struct device *dev, uint8_t *byte, uint32_t count)
+{
+	const struct bh1750_config *cfg = dev->config;
+	return i2c_read(cfg->bus, byte, count, cfg->bus_addr);
+}
+
+static int bh1750_write(const struct device *dev, uint8_t byte)
+{
+	const struct bh1750_config *cfg = dev->config;
+	return i2c_write(cfg->bus, &byte, 1, cfg->bus_addr);
+}
+
+static int bh1750_attr_set(const struct device *dev, enum sensor_channel chan,
+			   enum sensor_attribute attr, const struct sensor_value *val)
+{
+	int ret;
+	uint8_t byte;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_LIGHT || chan == SENSOR_CHAN_ALL);
+	__ASSERT_NO_MSG(attr == SENSOR_ATTR_OVERSAMPLING);
+
+	if (val->val1 < 31 || val->val1 > 254) {
+		LOG_ERR("Oversampling factor out of range [31, 254]: %d", val->val1);
+		return -EINVAL;
+	}
+
+	byte = BH1750_MEAS_TIME_MSB | (((uint8_t)val->val1 >> 5) & 0x07);
+	ret = bh1750_write(dev, byte);
+	if (ret < 0) {
+		LOG_ERR("Failed to configure high part of oversampling factor.");
```

Comment:
```suggestion
		LOG_ERR("Failed to configure %s part of oversampling factor.", "high");
```

You can save a few bytes if you do it like so, same goes for the low part below